mbuild/box.py

```diff
@@ -2,151 +2,333 @@
-        self._mins = _BoxArray(array=mins, var="mins", box=self)
-        self._lengths = _BoxArray(array=(self.maxs - self.mins), var="lengths", box=self)
-
-    @maxs.setter
-    def maxs(self, maxs):
-        maxs = np.array(maxs, dtype=np.float)
-        assert maxs.shape == (3, )
-        assert all(maxs >= self.mins), "Given maxs is less than mins"
-        self._maxs = _BoxArray(array=maxs, var="maxs", box=self)
-        self._lengths = _BoxArray(array= (self.maxs - self.mins), var="lengths", box=self)
-
-    @lengths.setter
-    def lengths(self, lengths):
-        if isinstance(lengths, int) or isinstance(lengths, float):
-            lengths = np.array(lengths*np.ones(3), dtype=np.float)
-        else:
-            lengths = np.array(lengths, dtype=np.float)
-        assert lengths.shape == (3, )
-        assert all(lengths >= 0), "Given lengths are negative" 
-        self._maxs = _BoxArray(array=(self.maxs + (0.5*lengths - 0.5*self.lengths)), var="maxs", box=self)
-        self._mins = _BoxArray(array=(self.mins - (0.5*lengths - 0.5*self.lengths)), var="mins", box=self)
-        self._lengths = _BoxArray(array=lengths, var="lengths", box=self, dtype=np.float)
-
-    @angles.setter
-    def angles(self, angles):
-        angles = np.array(angles, dtype=np.float)
-        assert angles.shape == (3, )
-        self._angles = _BoxArray(array=angles, var="angles", box=self, dtype=np.float)
+        return self._get_angles()
+
+    @property
+    def bravais_parameters(self):
+        """Return the Box representation as Bravais lattice parameters.
+
+        Based on the box vectors, return the parameters to describe the box in
+        terms of the Bravais lattice parameters:
+            a,b,c = the edges of the Box
+            alpha, beta, gamma = angles(tilt) of the parallelepiped, in degrees
+
+        Returns
+        -------
+        parameters : tuple of floats (a, b, c, alpha, beta, gamma)
+        """
+        (alpha, beta, gamma) = self._get_angles()
+        return (
+            self._Lx,
+            self._Ly,
+            self._Lz,
+            alpha,
+            beta,
+            gamma
+        )
 
     def __repr__(self):
-        return "Box(mins={}, maxs={}, angles={})".format(self.mins, self.maxs, self.angles)
+        (Lx, Ly, Lz, xy, xz, yz) = self.box_parameters
+        if self._precision is not None:
+            precision = self._precision
+            desc = (f"Box: Lx={Lx:.{precision}f}, "
+                    f"Ly={Ly:.{precision}f}, "
+                    f"Lz={Lz:.{precision}f}, "
+                    f"xy={xy:.{precision}f}, "
+                    f"xz={xz:.{precision}f}, "
+                    f"yz={yz:.{precision}f}")
+        else:
+            desc = (f"Box: Lx={Lx}, "
+                    f"Ly={Ly}, "
+                    f"Lz={Lz}, "
+                    f"xy={xy}, "
+                    f"xz={xz}, "
+                    f"yz={yz}")
```

Comment:
Simplified formatting:
```suggestion
        format_precision = f".{self._precision}f" if self._precision else ""
        desc = (f"Box: Lx={Lx:{format_precision}}, "
                f"Ly={Ly:{format_precision}}, "
                f"Lz={Lz:{format_precision}}, "
                f"xy={xy:{format_precision}}, "
                f"xz={xz:{format_precision}}, "
                f"yz={yz:{format_precision}}, ")
```